cpp/src/gandiva/basic_decimal_scalar.h

```diff
@@ -60,4 +68,21 @@ inline BasicDecimalScalar128 operator-(const BasicDecimalScalar128& operand) {
   return BasicDecimalScalar128{-operand.value(), operand.precision(), operand.scale()};
 }
 
+inline std::ostream& operator<<(std::ostream& os, const BasicDecimalScalar128& dec) {
+  os << dec.ToString();
+  return os;
+}
+
 }  // namespace gandiva
+
+namespace std {
+    template<>
+    struct hash<arrow::Decimal128> {
```

Comment:
Again, it is better to implement hash for gandiva::DecimalScalar instead of arrow::Decimal128. Implemnting hash for arrow::Decimal128 might conflict with any implemtations with arrow code later

Reply:
done